packages/identity-sdk/src/viem-sdk.ts

```diff
@@ -0,0 +1,215 @@
+import {
+  Address,
+  PublicClient,
+  WalletClient,
+  parseAbi,
+  SimulateContractParameters,
+  WalletActions,
+} from "viem";
+import { waitForTransactionReceipt } from "viem/actions";
+import { z } from "zod";
+import { compressToEncodedURIComponent } from "lz-string";
+
+import { Envs, FV_IDENTIFIER_MSG2 } from "./constants";
+
+const identityV2ABI = parseAbi([
+  "function addWhitelisted(address account)",
+  "function removeWhitelisted(address account)",
+  "function isWhitelisted(address account) view returns (bool)",
+  "function lastAuthenticated(address account) view returns (uint256)",
+  "function authenticationPeriod() view returns (uint256)",
+]);
+
+const addressSchema = z
+  .string()
+  .regex(/^0x[a-fA-F0-9]{40}$/, "Invalid Ethereum address");
+
+interface IdentityContract {
+  publicClient: PublicClient;
+  contractAddress: Address;
+  abi: any;
+}
+
+interface IdentityExpiryData {
+  lastAuthenticated: bigint;
+  authPeriod: bigint;
+}
+
+export const initializeIdentityContract = (
+  publicClient: PublicClient,
+  contractAddress: Address,
+): IdentityContract => ({
+  publicClient,
+  contractAddress,
+  abi: identityV2ABI,
+});
```

Comment:
contract address and abi is always the same, public client is passed in most of the methods.
the methods can instead receive at the end an optional `identityContractAddress: 'dev'|'staging'|'production'|0xstring  with default to production.`

Reply:
done